src/KafkaW/Msg.h

```diff
@@ -2,20 +2,31 @@
 
 #include <cstdint>
 #include <cstdlib>
+#include <functional>
 
 namespace KafkaW {
 // Want to expose this typedef also for users of this namespace
 using uchar = unsigned char;
 
 class Msg {
 public:
+  Msg() = default;
+  Msg(std::uint8_t const *Pointer, size_t Size,
+      std::function<void()> DataDeleter)
+      : DataPointer(Pointer), DataSize(Size), OnDelete(DataDeleter) {}
+  Msg(std::uint8_t const *Pointer, size_t Size,
+      std::function<void()> DataDeleter, std::int64_t Offset)
+      : DataPointer(Pointer), DataSize(Size), OnDelete(DataDeleter),
+        MessageOffset(Offset) {}
```

Comment:
There is an implicit cast of `Offset` from signed to unsigned here, best to make explicit with `static_cast` if intentional.

Reply:
My mistake. Was supposed to use `std::int64_t` as the member variable. This has now been fixed.